### PR TITLE
Support v-slot

### PR DESCRIPTION
**Pull request by lmiller1990**
Working on https://github.com/vuejs/vue-test-utils/issues/1261

- support the new `v-slot` syntax!
- need to bump to Vue v2.6
- likely a component combining the new and old scoped syntax will fail; we need to cover that case, it should be pretty easy, though.
- will need to rebase and fix my commits, the commit convention is wrong
- not sure if this is the best place to do this logic; likely we should have a simple util function to handle all the null checks.
- added two tests; one with SFC, one without. Likely we can nuke one... just wanted to be thorough.

### CODE REVIEW

**Comment by JessicaSachs** [comment]
On the surface, this looks good to me. Like you said, we probably want to refactor it into a function to hide the nested checks... but aside from that 👍 

I'm curious if $parent will always be present -- can root elements have scopedSlots? I'm particularly thinking of components constructed with createElement.

**Comment by lmiller1990** [comment]
@JessicaSachs will need to investigate more to answer you questions, but got around to doing a refactor.

My commit messages aren't styled property, will use the squash feature on Github to have a single nice commit message when we merge this. Might even be fine to merge as is and come back to address - thoughts? Maybe we don't cover every edge case, but should at least cover some.

File: packages/create-instance/create-component-stubs.js [hunk 1]
```diff
@@ -75,6 +81,24 @@ function resolveOptions(component, _Vue) {
...
+  // We filter out the _normalized and $stable key
+  if (
+    ctx &&
+    ctx.$options &&
+    ctx.$options.parent._vnode &&
+    ctx.$options.parent._vnode.data &&
+    ctx.$options.parent._vnode.data.scopedSlots
+  ) {
+    const slotKeys: Array<string> = ctx.$options.parent._vnode.data.scopedSlots
+    return keys(slotKeys).filter(x => x[0] !== '_' && !x.includes('$'))
```
Comment (afontcu) [inline packages/create-instance/create-component-stubs.js]: Thinking out loud here, and might be missing something, but:

If we're aiming to remove `_normalized` and `$stable`,  could we be more explicit about our intention?

```js
return keys(slotKeys).filter(x => x !== '_normalized' && x !== '$stable')
```

(or even `['_normalized', '$stable'].includes(x)`)

Comment (lmiller1990) [reply packages/create-instance/create-component-stubs.js]: Yep, sounds good. I don't think there should be any other special variables beginning with `_` or `$`.

**Review by afontcu — APPROVED** [review]

**Comment by lmiller1990** [comment]
@JessicaSachs I tried a component like this:

```vue
<template>
  <slot name="blah" />
</template>
```

To see what would happen to `parent`, since `parent` should be `undefined`. Turns out you cannot do this, Vue will throw an error:

```
(Emitted value instead of an instance of Error)
    Error compiling template:

    <slot name="newSyntax" />

    - Cannot use <slot> as component root element because it may contain multiple nodes.
```

I added a check for good measure, just in case.

**Comment by JessicaSachs** [comment]
@lmiller1990 yeah you can’t make slots the root element when you use template functions, but you can with createElement and a render function.

**Comment by lmiller1990** [comment]
Hm, maybe I am missing something - does it make sense for the root element (eg `App.vue` most of the time) to have a `<slot />` in the first place? That would imply you are passing something from above the root to be rendered in the slot.

I couldn't figure out how to write a render function with slots - if you can write a basic example I would be happy to add a test and see what happens.

Thanks for the thorough review... slots feel like the most edge-casey of anything in VTU. We definitely need to get this right.

**Review by JessicaSachs — APPROVED** [review]
I think you're right. This looks good.

I spent a few hours trying to break this using render functions on root vue components and other use cases and couldn't get it to break. I do a bunch of edgecase-y things in my prod app and this seems to handle all of it.

**Review by dobromir-hristov — APPROVED** [review]

**Comment by afontcu** [comment]
Great job mate! 🙌

**Comment by lmiller1990** [comment]
Going to add one more test around `v-slot` then merge this up, should be done by the end of the week
